uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/manager-leave/components/manager-leave-requests_test.js

```diff
@@ -1,30 +1,895 @@
+          });
+        });
+      });
+
+      describe('pagination', function () {
+
+        describe('totalNoOfPages', function () {
+
+          var returnValue;
+
+          beforeEach(function () {
+            controller.leaveRequests.table.total = 20;
+            controller.pagination.size = 5;
+            returnValue = controller.totalNoOfPages();
+          });
+
+          it('returns correct number of pages', function () {
+            expect(returnValue).toBe(4);
+          });
+        });
+
+        describe('nextPage', function () {
+
+          beforeEach(function () {
+            spyOn(controller, 'totalNoOfPages').and.returnValue(2);
+            spyOn(controller, 'refresh');
+          });
+
+          describe('current page is less than total page number', function () {
+
+            beforeEach(function () {
+              controller.pagination.page = 1;
+              controller.totalNoOfPages.and.returnValue(2);
+              controller.nextPage();
+            });
+
+            it('page no gets increased', function () {
+              expect(controller.pagination.page).toBe(2);
+            });
+
+            it('calls refresh', function () {
+              expect(controller.refresh).toHaveBeenCalledWith(2);
+            });
+          });
+
+          describe('current page is not less than total page number', function () {
+
+            beforeEach(function () {
+              controller.pagination.page = 3;
+              controller.totalNoOfPages.and.returnValue(3);
+              controller.nextPage();
+            });
+
+            it('does not increase page no', function () {
+              expect(controller.pagination.page).toBe(3);
+            });
+
+            it('calls refresh', function () {
+              expect(controller.refresh).not.toHaveBeenCalled();
+            });
+          });
+        });
+      });
+
+      describe('period label', function () {
+        var label, period;
+
+        describe('when the period is current', function () {
+          beforeEach(function () {
+            period = _(controller.absencePeriods).find(function (period) {
+              return period.current;
+            });
+            label = controller.labelPeriod(period);
+          });
+
+          it('labels it as such', function () {
+            expect(label).toBe('Current Period (' + period.title + ')');
+          });
+        });
+
+        describe('when the period is not current', function () {
+          beforeEach(function () {
+            period = _(controller.absencePeriods).filter(function (period) {
+              return !period.current;
+            }).sample();
+            label = controller.labelPeriod(period);
+          });
+
+          it('returns the title as it is', function () {
+            expect(label).toBe(period.title);
+          });
+        });
+      });
+
+      describe('getLeaveStatusByValue', function () {
+
```

Comment:
I have kept the single line gaps after every describe block intentionally, to give some breathing space.